modules/lib/src/main/resources/assets/js/app/AppWrapper.ts

```diff
@@ -144,85 +107,58 @@ export class AppWrapper
-        return widget === this.widgets[0];
+        return widget === Store.$sidebarWidgets.get().items?.[0];
     }
 
-
     private fetchAndAppendWidget(widget: Widget): void {
-        if (this.isDefaultWidget(widget)) { // default studio app
+        if (this.isDefaultWidget(widget)) {
+            // default studio app
             const widgetEl: Element = this.createStudioWidgetEl();
             this.widgetElements.set(widget.getWidgetDescriptorKey().toString(), {el: widgetEl});
             this.widgetsBlock.appendChild(widgetEl);
             return;
         }
 
         fetch(widget.getFullUrl())
-            .then(response => response.text())
+            .then((response) => response.text())
             .then((html: string) => {
-                WidgetHelper.createFromHtmlAndAppend(html, this.widgetsBlock)
-                    .then((widgetEl: WidgetElement) => {
-                        const widgetKey = widget.getWidgetDescriptorKey().toString();
-                        this.widgetElements.set(widgetKey, widgetEl);
-                        this.activeWidgets.push(widgetKey);
-                    });
+                WidgetHelper.createFromHtmlAndAppend(html, this.widgetsBlock).then((widgetEl: WidgetElement) => {
+                    const widgetKey = widget.getWidgetDescriptorKey().toString();
+                    this.widgetElements.set(widgetKey, widgetEl);
+                    this.activeWidgets.push(widgetKey);
+                });
             })
-            .catch(err => {
+            .catch((err) => {
                 throw new Error('Failed to fetch widget: ' + err);
             });
     }
 
-    private updateSidebarWidgets() {
-        new GetWidgetsByInterfaceRequest('contentstudio.menuitem').sendAndParse().then((widgets: Widget[]) => {
-            widgets.push(this.widgets[0]); // default studio widget
-            this.removeStaleWidgets(widgets);
-            this.addMissingWidgets(widgets);
-        }).catch(DefaultErrorHandler.handle);
-    }
-
-    private removeStaleWidgets(widgets: Widget[]): void {
-        this.widgets = this.widgets.filter((currentWidget: Widget) => {
-            if (widgets.some((w: Widget) => w.getWidgetDescriptorKey().equals(currentWidget.getWidgetDescriptorKey()))) {
-                return true;
-            }
-
-            this.sidebar.removeWidget(currentWidget);
-
-            return false;
-        });
-    }
-
-    private hasWidget(widget: Widget): boolean {
-        return this.widgets.some((w: Widget) => w.getWidgetDescriptorKey().equals(widget.getWidgetDescriptorKey()));
-    }
-
-    private addMissingWidgets(widgets: Widget[]): void {
-        widgets.filter((widget: Widget) => !this.hasWidget(widget)).forEach((widget: Widget) => {
-            this.widgets.push(widget);
-            this.sidebar.addWidget(widget);
-            this.notifyItemAdded(widget);
-        });
+    private loadWidgets() {
+        Store.loadWidgets().catch(DefaultErrorHandler.handle);
```

Comment:
Can be removed entirely (see the comments in the store for details).
The `catch` block won’t catch anything, since you’re not throwing a new error and already have `try–catch–finally` blocks inside.